packages/edit-site/src/index.js

```diff
@@ -41,7 +41,7 @@ export async function reinitializeEditor( target, settings ) {
 	// won't be present. Do a client side redirect to the 'homepage' if that's
 	// the case.
 	try {
-		await redirectToHomepage( settings.siteUrl );
+		await redirectToHomepage( settings.__experimentalHomeTemplate );
```

Comment:
I wonder if we could just do the redirection on server-side instead, so we don't need to do anything on the client side (except for exception handling I guess).